kuma/wiki/management/commands/render_document.py

```diff
@@ -2,100 +2,115 @@
+    args = '<document_path document_path ...>'
+    help = 'Render a wiki document'
     option_list = BaseCommand.option_list + (
-        make_option('--all', dest="all", default=False,
-                    action="store_true",
-                    help="Render ALL documents"),
-        make_option('--min-age', dest="min_age", default=600,
-                    help="Documents rendered less than this many seconds ago "
-                         "will be skipped"),
-        make_option('--baseurl', dest="baseurl",
+        make_option('--all', dest='all', default=False, action='store_true',
+                    help='Render ALL documents'),
+        make_option('--min-age', dest='min_age', default=600,
+                    help='Documents rendered less than this many seconds ago '
+                         'will be skipped'),
+        make_option('--baseurl', dest='baseurl', default=False,
+                    help='Base URL to site'),
+        make_option('--force', action='store_true', dest='force',
                     default=False,
-                    help="Base URL to site"),
-        make_option('--force', action="store_true", dest="force",
+                    help='Force rendering, first clearing record of any '
+                         'rendering in progress'),
+        make_option('--nocache', action='store_true', dest='nocache',
                     default=False,
-                    help="Force rendering, first clearing record of any "
-                         "rendering in progress"),
-        make_option('--nocache', action="store_true", dest="nocache",
+                    help='Use Cache-Control: no-cache instead of max-age=0'),
+        make_option('--defer', action='store_true', dest='defer',
                     default=False,
-                    help="Use Cache-Control: no-cache instead of max-age=0"),
-        make_option('--defer', action="store_true", dest="defer",
-                    default=False,
-                    help="Defer rendering"),
+                    help='Defer rendering by chaining tasks via celery'),
     )
 
     def handle(self, *args, **options):
         self.options = options
         self.base_url = options['baseurl'] or absolutify('')
+        if self.options['nocache']:
+            self.cache_control = 'no-cache'
+        else:
+            self.cache_control = 'max-age=0'
 
         if options['all']:
-            logging.info(u"Querying ALL %s documents..." %
-                         Document.objects.count())
-            docs = Document.objects.order_by('-modified').iterator()
-            for doc in docs:
-                self.do_render(doc)
+            # Query all documents, excluding those whose `last_rendered_at` is
+            # within `min_render_age` or NULL.
+            min_render_age = (
+                datetime.datetime.now() -
+                datetime.timedelta(seconds=self.options['min_age']))
+            docs = Document.objects.filter(
```

Comment:
Is there a model manager method or model method that returns this type of query already?